### PR TITLE
Tidy up GitHub Actions

### PR DESCRIPTION
**Pull request by mudge**
Try to speed up builds by reducing duplication (no longer testing the Rake extension task but only testing the gems).

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -7,19 +7,68 @@ on:
...
+          path: gems
+
   compile-and-test-system-dependencies:
+    name: Ruby ${{ matrix.ruby }} - libre2.${{ matrix.libre2.soname }}
     needs: ["build-cruby-gem"]
-    name: System Dependencies - ${{ matrix.sys }} vendored libs - Ruby ${{ matrix.ruby }} - libre2 ABI version ${{ matrix.libre2.soname }}
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        sys: ["enable", "disable"]
```
Comment (mudge) [inline .github/workflows/tests.yml:22]: @stanhu do you have any thoughts about pulling this seeing as we have `compile-and-test-vendored-dependencies` which _should_ cover the same cases (compiling the gem and its vendored dependencies on all supported platforms)?

Comment (stanhu) [reply .github/workflows/tests.yml:22]: As I recall, I added this in https://github.com/mudge/re2/pull/93 which covers the case when system libraries are installed, but the gem wants to use the vendored libraries and compile from source.

Does one of the checks in https://github.com/mudge/re2/actions/runs/6253089644/job/16977764503 cover this case?

Comment (mudge) [reply .github/workflows/tests.yml:22]: Ah, let me add a specific test case for that: using the vendored libraries but with libre2 installed in Ruby's `exec_prefix` (I guess we don't need to exhaustively test that against every platform and Ruby version?)

Comment (mudge) [reply .github/workflows/tests.yml:22]: I've added a dedicated job for testing the installation of the gem (with the default `---disable-system-libraries`) but where there is already a libre2 in the Ruby `exec_prefix` (and a system version of Abseil installed).

Comment (mudge) [reply .github/workflows/tests.yml:22]: I've updated this to use whatever `libre2-dev` is available on `ubuntu-latest` but still symlinking it into Ruby's `lib` directory.

**Comment by mudge** [comment]
@stanhu this shaves a significant amount of time from a build down (down from over 30 minutes in some cases to ~15 minutes): are you happy we're not losing any important coverage by testing installing the gem (with and without `--enable-system-libraries`) rather than testing `rake compile spec` directly?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (stanhu) [reply .github/workflows/tests.yml:22]: Sounds good.

**Comment by stanhu** [comment]
>  are you happy we're not losing any important coverage by testing installing the gem (with and without --enable-system-libraries) rather than testing rake compile spec directly?

Yes, that sounds good.

**Pull request merged**